src/Resources/assets/shop/js/copyToWishlistsListModal.js

```diff
@@ -0,0 +1,128 @@
+    }
+
+    _modalTemplate() {
+        const modal = document.createElement('div');
+        modal.innerHTML = `    
+        <form class=${this.finalConfig.confirmationModalClass}>
+            <header class=${this.finalConfig.confirmationModalheaderClass}>
+                <h2 class=${this.finalConfig.confirmationModalH2Class}>
+                    ${this.finalConfig.headerTitle}
+                </h2>
+            </header>
+            <section data-bb-target="wishlists" class=${this.finalConfig.confirmationModalBodyClass}>
+                    
+            </section>
+            <section class=${this.finalConfig.confirmationModalConfirmClass}>
+                <button type="button" data-bb-action="cancel" class=${this.finalConfig.confirmationModalCancelBtnClass}>
+                    ${ this.finalConfig.cancelText }
+                </button>
+                <button type="button" data-bb-action="perform" class=${this.finalConfig.confirmationModalConfirmBtnClass}>
+                    ${this.finalConfig.performText}
+                </button>
+            </section>
+        </form>
+        `;
+        modal.querySelector('[data-bb-target="wishlists"]').appendChild(this._getWishlists());
+
+        return modal;
+    }
+
+    _wishlistTemplate(wishlist) {
+        return `    
+            <option value="${wishlist.dataset.bbWishlists}" data-bb-wishlist-id="${wishlist.dataset.bbWishlistsId}">
+                ${wishlist.dataset.bbWishlists}
+            </option>
+        `;
+    }
+
+    _getWishlists() {
+        const select = document.createElement('select')
+        select.name = 'wishlist'
+        select.classList.add('ui', 'selection', 'dropdown')
+        select.insertAdjacentHTML("beforeend" ,'<option selected disabled>My Wishlist…</option>')
+
+        this.wishlistTargets.forEach(wishlist => {
+            select.insertAdjacentHTML("beforeend" , this._wishlistTemplate(wishlist))
+        });
+        return select;
+    }
+
+    _matchWishlists(targetWishlists) {
+        this.wishlistTargetsId.forEach(wishlist => {
+            if (wishlist.dataset.bbWishlists == targetWishlists.value) {
+                this.copyTarget = wishlist.dataset.bbWishlistsId
+            }
+        });
+    }
+
+    _modalActions(template) {
+        const cancelBtn = template.querySelector('[data-bb-action="cancel"]');
+        const confirmBtn = template.querySelector('[data-bb-action="perform"]');
+
+        cancelBtn.addEventListener('click', (e) => {
+            e.preventDefault();
+            this.actions.cancelAction();
+            this._closeModal();
+        });
+
+        confirmBtn.addEventListener('click', (e) => {
+            e.preventDefault();
+            this._matchWishlists(document.querySelector('[data-bb-target] select'));
+            this.actions.performAction(this.wishlistcurrent, this.copyTarget);
+            this._closeModal();
+        });
+    }
+
+    _closeModal() {
+        this.modal.remove();
+    }
+}
+
```

Comment:
AddToWIshlistModal - CreateWishlistModal - cannot these 2 class be one? just change their config when invoking them by providing config object as parameter eg new class = new AddToWishlistModal({class1=""},{cancelAction:()=>{{}, performAction: ()=>{})?

Also names - class should be name by its function, not by its use case - so WishlistModal - instead of AddWishlistModal or CreateWishlsitModal.

If class is designed to be reused it can be done, but these classes are changed inside, i would consider their names change.